### PR TITLE
[JENKINS-63835] Added BranchName/CommitHash as optional parameters

### PR DESCRIPTION
**Pull request by RiederAlex**

### CODE REVIEW

**Comment by mezpahlan** [comment]
Hi @RiederAlex Apologies for the delay. I'm mega busy at work this week. I'll try and find some time to look over this shortly. Sorry again.

File: src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java [hunk 1]
```diff
@@ -99,8 +101,15 @@ private String parseReleaseNotesFromFile(@Nonnull String pathToReleaseNotes) {
...
+    private static BuildInfo CreateBuildInfo(@Nonnull UploadRequest request) {
+        if (request.branchName == null && request.commitHash == null) return null;
+        return new BuildInfo(request.branchName,request.commitHash,null);
+    }
+
     @Override
     public PrintStream getLogger() {
         return taskListener.getLogger();
     }
+
```
Comment (mezpahlan) [inline src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java:113]: Just a small comment, but I think we should remove this blank space. At the end of this review I'll do this if there aren't other comments that I need your help with.

File: src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java [hunk 2]
```diff
@@ -99,8 +101,15 @@ private String parseReleaseNotesFromFile(@Nonnull String pathToReleaseNotes) {
 
     }
 
+    private static BuildInfo CreateBuildInfo(@Nonnull UploadRequest request) {
```
Comment (mezpahlan) [inline src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java:104]: Does this need to be static? I'm guessing not.

Comment (mezpahlan) [inline src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java:104]: I think we should change CreateBuildInfo to createBuildInfo as (in so far as I can tell) it is simply a private method of this class.... not a class itself.

Comment (mezpahlan) [inline src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java:104]: We should also add the appropriate nullability information here. i.e. `@Nullable`

File: src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java [hunk 3]
```diff
@@ -99,8 +101,15 @@ private String parseReleaseNotesFromFile(@Nonnull String pathToReleaseNotes) {
...
+        if (request.branchName == null && request.commitHash == null) return null;
+        return new BuildInfo(request.branchName,request.commitHash,null);
```
Comment (mezpahlan) [inline src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java:106]: Just a quick question do you know what will happen if the user supplies _both_ the branch name and the commit hash? Is that possible? Allowed? Should we protect against this earlier in the transaction?

Comment (mezpahlan) [reply src/main/java/io/jenkins/plugins/appcenter/task/internal/DistributeResourceTask.java:106]: Ahh ignore me this seems only be purely informational like the release notes. So I guess this won't fail but will be confusing for the user. But then that's their fault. Hahaha!

**Review by mezpahlan — CHANGES_REQUESTED** [review]
I think we should tidy up some small minor points but otherwise really nice PR. Thank you very much for this. And apologies for the delay in getting back to you.

**Comment by mezpahlan** [comment]
Hi @RiederAlex I've resolved the comments in the PR in a local branch of mine but am unable to push them to your repo. In the PR do you have the option to allow maintainers (i.e. me) to submit upates to your branch? And if so can you enable that option please.

**Comment by RiederAlex** [comment]
> Hi @RiederAlex I've resolved the comments in the PR in a local branch of mine but am unable to push them to your repo. In the PR do you have the option to allow maintainers (i.e. me) to submit upates to your branch? And if so can you enable that option please.

Sadly I just learned that this is not possible directly, as I have forked it as a company account and not as a user account - so I gave you write access directly to the fork - I'm sorry for the inconvenience and thanks for the review!

**Comment by mezpahlan** [comment]
@RiederAlex Thanks, not a problem. I've pushed my changes and merged in the latest main branch. It's just building now.

**Pull request merged**